include/LeMonADE/feature/FeatureNNInteractionBcc.h

```diff
@@ -0,0 +1,602 @@
+
+This file is part of LeMonADE.
+
+LeMonADE is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+LeMonADE is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
+
+--------------------------------------------------------------------------------*/
+
+
+#ifndef FEATURE_NN_INTERACTION_BCC_H
+#define FEATURE_NN_INTERACTION_BCC_H
+
+/**
+ * @file
+ * @date 2016/06/18
+ * @author Hauke Rabbel
+ * @brief Definition and implementation of class template FeatureNNInteractionBcc
+ * 
+ * @todo MoveAddBccMonomer is used here, which might be obsolete. 
+**/
+
+#include <LeMonADE/feature/Feature.h>
+#include <LeMonADE/updater/moves/MoveBase.h>
+#include <LeMonADE/updater/moves/MoveLocalBcc.h>
+#include <LeMonADE/updater/moves/MoveAddBccMonomer.h>
+#include <LeMonADE/feature/FeatureExcludedVolumeBcc.h>
+#include <LeMonADE/feature/FeatureBoltzmann.h>
+#include <LeMonADE/feature/FeatureAttributes.h>
+#include <LeMonADE/feature/FeatureNNInteractionReadWrite.h>
+
+/**
+ * @class FeatureNNInteractionBcc
+ * @brief Provides interaction of monomers on distances d<=sqrt(12)a for bccBFM
+ *
+ * @tparam FeatureLatticeType Underlying lattice feature, e.g. FeatureLattice or 
+ * FeatureLatticePowerOfTwo (template template parameter)
+ *
+ * @details
+ * The interaction energy can be set for pairs of monomer-types A,B, where
+ * the type can be any integer between 1 and 255. 
+ * The feature automatically adds FeatureExcludedVolumeBcc<FeatureLatticeType<uint8_t>
+ * to the system. Given an energy E in kT between  two types, the interaction potential 
+ * as a function of the distance d is:
+ * - inf d < sqrt(3) (implicitly through excluded volume)
+ * - E sqrt(3)< d <= sqrt(12)
+ * - 0   d>sqrt(12)
+ * .
+ * Usage: In the feature list defining Ingredients use this feature as
+ * FeatureNNInteractionBcc<FeatureLattice> (arbitrary lattices), or as
+ * FeatureNNInteractionBcc<FeatureLatticePowerOfTwo> (2**n lattices)
+ * The feature adds the bfm-file command !nn_interaction A B E
+ * for monomers of types A B with interaction energy of E in kT.
+**/
+
+template<template<typename> class FeatureLatticeType>
+class FeatureNNInteractionBcc:public Feature
+{
+
+private:
+  //! Type for the underlying lattice, used as template parameter for FeatureLatticeType<...>
+  typedef uint8_t lattice_value_type;
+
+  //! Interaction energies between monomer types. Max. type=255 given by max(uint8_t)=255 
+  double interactionTable[256][256];
```

Comment:
I would like to use std::containers, but this should not change the whole thing. But for instance you do not have to allocate memory explicitly, see constructor.